### PR TITLE
[Builder] Move `builder.py` to `api/utils/builder.py` 

### PR DESCRIPTION
**Pull request by Tankilevitch**
Part of the client server separation moved builder under the API hierarchy as it is only used by the API

### CODE REVIEW

File: mlrun/runtimes/kubejob.py [hunk 1]
```diff
@@ -240,17 +240,6 @@ def deploy(
                 state = self._build_watch(watch, show_on_failure=show_on_failure)
                 ready = state == "ready"
                 self.status.state = state
-        else:
```
Comment (alonmr) [inline mlrun/runtimes/kubejob.py:243]: 🔥 
I plan to move the deploy to the launchers and have it raise an exception if called from the server side FYI

**Review by alonmr — COMMENTED** [review]
minor

File: mlrun/runtimes/kubejob.py (hunk 1 above)
Comment (Tankilevitch) [reply mlrun/runtimes/kubejob.py:243]: With _run it is more obvious why to the runtime handler, but for the deploy there is a debate, I suggest we discuss it offline

File: mlrun/runtimes/utils.py [hunk 2]
```diff
@@ -371,7 +375,7 @@ def resolve_function_target_image_registries_to_enforce_prefix():
     registry, repository = helpers.get_parsed_docker_registry()
     repository = helpers.get_docker_repository_or_default(repository)
     return [
-        f"{mlrun.builder.IMAGE_NAME_ENRICH_REGISTRY_PREFIX}{repository}/",
+        f"{mlrun.api.utils.builder.IMAGE_NAME_ENRICH_REGISTRY_PREFIX}{repository}/",
```
[the remark this answers is not in the document]
Comment (Tankilevitch) [reply mlrun/runtimes/utils.py:378]: agreed, will move

**Pull request merged**